lib/cell.rb

```diff
@@ -23,7 +24,24 @@ def fired_upon?
 
   def fire_upon
     @fired_upon = true
-    !empty?
-    ship.hit
+    if !empty?
+      ship.hit
+    end
+    
   end
-end
\ No newline at end of file
+
+  def render
+    if fired_upon? && empty?
```

Comment:
nice work! we'll definitely build out the rest of the conditionals this way in the render method.  we'll probably create some helpers to help dry some of it up too. great work !